### PR TITLE
Fix the crash for projects using JVM 17 #367

### PR DESCRIPTION
**Pull request by OmarDefaoui**
Fix the crash for projects using JVM 17 #367

### CODE REVIEW

**Comment by Zazo032** [comment]
@OmarDefaoui thanks for the fix! Could this also be solved by upgrading `ext.kotlin_version` (last is `1.9.20`) and `com.android.tools.build:gradle` (last is `8.1.3`)?

**Comment by OmarDefaoui** [comment]
Yes, it will also fix the issue

**Comment by vlazdra** [comment]
Hey @Zazo032 the project is not building if I don't add this inside the `android` blcok:
```gradle
    compileOptions {
      sourceCompatibility JavaVersion.VERSION_17
      targetCompatibility JavaVersion.VERSION_17
    }
```

When I add it, it then works and builds my project.
Note: I've updated the kotlin plugin to 1.9.20 and AGP to 8.1.4 for both my project and the `intercom_flutter` package.

**Comment by deepak786** [comment]
> Hey @Zazo032 the project is not building if I don't add this inside the `android` blcok:
> 
> ```groovy-gradle
>     compileOptions {
>       sourceCompatibility JavaVersion.VERSION_17
>       targetCompatibility JavaVersion.VERSION_17
>     }
> ```
> 
> When I add it, it then works and builds my project. Note: I've updated the kotlin plugin to 1.9.20 and AGP to 8.1.4 for both my project and the `intercom_flutter` package.

I don't think that this change should be added in intercom_flutter. It must be added in root project/app who is using the intercom_flutter and updated Kotlin plugin and AGP.

**Comment by vlazdra** [comment]
Hey @deepak786. 👋 

If I remove the `compileOptions` block from the `intercom_flutter` source code, I get this error:
```bash
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':intercom_flutter:compileDebugKotlin'.
> Inconsistent JVM-target compatibility detected for tasks 'compileDebugJavaWithJavac' (1.8) and 'compileDebugKotlin' (17).

  Consider using JVM Toolchain: https://kotl.in/gradle/jvm/toolchain
  Learn more about JVM-target validation: https://kotl.in/gradle/jvm/target-validation 
```

But if I add the compile block back, it works.
Maybe I'm doing something wrong? Do you maybe have some idea?

**Comment by deepak786** [comment]
Can you try by adding the compile options block in your project build.gradle file itself instead of intercom_flutter?

**Comment by vlazdra** [comment]
I did. I mean, it's there the whole time.
Screenshot 2023-11-20 at 17 09 42

Maybe a wild guess, maybe the under-the-hood intercom android sdk has the 1.8 fixed in their build.gradle? Don't know since it's closed source.

**Comment by deepak786** [comment]
@vlazdra Please send a query to Intercom support directly.

**_Note: Don't mention anything about Flutter. Raise the query considering the Android platform only._**

**Comment by vlazdra** [comment]
@deepak786 Will do.

No worries I'm aware that they won't assist for something that they don't support officially.

I did however point out, during a meeting, that they should officially support it since it's a major player in the market and it's just a matter of "bridging the gap". 🙂

**Comment by deepak-agendaboa** [comment]
```
compileOptions {
    sourceCompatibility JavaVersion.VERSION_1_8
    targetCompatibility JavaVersion.VERSION_1_8
}

kotlinOptions {
    jvmTarget = '1.8'
}
```

Adding this will also fix the issue.

**Comment by deepak786** [comment]
```
compileOptions {
    sourceCompatibility JavaVersion.VERSION_1_8
    targetCompatibility JavaVersion.VERSION_1_8
}

kotlinOptions {
    jvmTarget = '1.8'
}
```

This will also solve the issue. 
We should update with this. 
Firebase packages are also using this e.g https://github.com/firebase/flutterfire/blob/eb9c7874dc7e34afba1c2fd0c02c6d56bd6adfc0/packages/firebase_core/firebase_core/android/build.gradle#L36

File: intercom_flutter/android/build.gradle [hunk 1]
```diff
@@ -39,6 +39,15 @@ android {
...
     }
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
+    }
+
+    kotlinOptions {
+        jvmTarget = JavaVersion.VERSION_17
+    }
```
Comment (deepak786) [inline intercom_flutter/android/build.gradle:50]: Update to

```
compileOptions {
    sourceCompatibility JavaVersion.VERSION_1_8
    targetCompatibility JavaVersion.VERSION_1_8
}

kotlinOptions {
    jvmTarget = '1.8'
}
```

Ref: https://github.com/firebase/flutterfire/blob/eb9c7874dc7e34afba1c2fd0c02c6d56bd6adfc0/packages/firebase_core/firebase_core/android/build.gradle#L36

**Comment by deepak786** [comment]
I have created another PR https://github.com/v3rm0n/intercom_flutter/pull/387 to add full support of AGP 8

**Comment by deepak786** [comment]
Try `intercom_flutter: 8.0.7`

**Comment by vlazdra** [comment]
> Try `intercom_flutter: 8.0.7`

@deepak786 Just tried it, and it seams that it's working as expected.

**Pull request closed without merging**